frontend/storm_wiki/pages_util/MyArticles.py

```diff
@@ -0,0 +1,187 @@
+import os
+import logging
+import streamlit as st
+from util.file_io import FileIOHelper
+from util.ui_components import UIComponents
+from util.theme_manager import load_and_apply_theme
+from db.db_operations import load_setting, save_setting
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+def update_num_columns():
+    save_setting("num_columns", st.session_state.num_columns_input)
+
+
+def ensure_default_categories():
+    categories = load_setting("categories", ["Default", "Uncategorized"])
+    for category in categories:
+        local_dir = FileIOHelper.get_output_dir(category)
+        os.makedirs(local_dir, exist_ok=True)
+    save_setting("categories", categories)
+    return categories
+
+
+def initialize_session_state():
+    if "page_size" not in st.session_state:
+        st.session_state.page_size = 24
+    if "current_page" not in st.session_state:
+        st.session_state.current_page = 1
+    if "selected_category" not in st.session_state:
+        st.session_state.selected_category = "All Categories"
+    if "num_columns" not in st.session_state:
+        st.session_state.num_columns = load_setting("num_columns", 3)
+
+    categories = ensure_default_categories()
+
+    if "user_articles" not in st.session_state:
+        st.session_state.user_articles = {}
+        for category in categories:
+            local_dir = FileIOHelper.get_output_dir(category)
+            st.session_state.user_articles[category] = (
+                FileIOHelper.read_structure_to_dict(local_dir)
+            )
+
+
+def get_all_articles():
+    all_articles = {}
+    for category, articles in st.session_state.user_articles.items():
+        for article_key, article_data in articles.items():
+            all_articles[f"{category}/{article_key}"] = article_data
+    return all_articles
+
+
+def display_article_list(page_size, num_columns):
```

Comment:
In my test, I can create new article successfully. But "My Articles" does not display properly, it just shows "Showing all 0 articles". I am not sure whether I mess up configuration somewhere.

Also, by reading the code, I have a question: Since `st.session_state` will be reset every time you restart the streamlit app, how are the articles maintained? Where can I find them?

Reply:
it is handled with https://github.com/jaigouk/storm_wiki/blob/main/pages_util/MyArticles.py#L55
`FileIOHelper.get_output_dir(category)` used and ` we [load articles](https://github.com/jaigouk/storm_wiki/blob/main/pages_util/MyArticles.py#L42) 

I will resolve issues in the weekend. thanks for the feedbacks.